semtypes/src/main/java/io/ballerina/types/Core.java

```diff
@@ -286,11 +286,11 @@
                     }
                 }
             }
-            if (data instanceof AllOrNothingSubtype && ((AllOrNothingSubtype) data).isAllSubtype()) {
+            if (!(data instanceof AllOrNothingSubtype)) {
```

Comment:
Let's add a comment to indicate that this is change to reflect the changes done for JBUG